test/core/util/test_lb_policies.cc

```diff
@@ -831,55 +830,20 @@ class QueueOnceLoadBalancingPolicy : public ForwardingLoadBalancingPolicy {
       parent_->channel_control_helper()->AddTraceEvent(severity, message);
     }
 
-    void SetPickQueued() {
-      MutexLock lock(&mu_);
-      pick_queued_ = true;
-      if (state_to_update_.has_value()) {
-        // Propagate the update from PickFirst but do not do it inline to avoid
-        // deadlocks.
-        parent_->work_serializer()->Schedule(
-            [helper = parent_->channel_control_helper(),
-             update_state = std::move(state_to_update_.value())]() {
-              helper->UpdateState(update_state.state, update_state.status,
-                                  update_state.picker);
-            },
-            DEBUG_LOCATION);
-        // Since we scheduled the update above instead of running inline, we
-        // need to drain the queue for the WorkSerializer.
-        ExecCtx::Run(
-            DEBUG_LOCATION,
-            GRPC_CLOSURE_CREATE(
-                [](void* arg, grpc_error_handle /*error*/) {
-                  auto parent = RefCountedPtr<QueueOnceLoadBalancingPolicy>(
-                      static_cast<QueueOnceLoadBalancingPolicy*>(arg));
-                  parent->work_serializer()->DrainQueue();
-                },
-                parent_->Ref().release(), nullptr),
-            absl::OkStatus());
-        state_to_update_.reset();
-      }
-    }
-
    private:
-    Mutex mu_;
-    struct StateToUpdate {
-      grpc_connectivity_state state;
-      absl::Status status;
-      RefCountedPtr<SubchannelPicker> picker;
-    };
-    absl::optional<StateToUpdate> state_to_update_ ABSL_GUARDED_BY(mu_);
-    bool pick_queued_ ABSL_GUARDED_BY(mu_) =
-        false;  // Has a pick been queued yet
     RefCountedPtr<QueueOnceLoadBalancingPolicy> parent_;
   };
+  struct StateToUpdate {
+    grpc_connectivity_state state;
+    absl::Status status;
+    RefCountedPtr<SubchannelPicker> picker;
+  };
+  absl::optional<StateToUpdate> state_to_update_;
```

Comment:
I don't think we need to use `absl::optional<>` here.  There shouldn't be any case where we use these values where they are not already set.

Reply:
You're right. I originally wanted to push the update in the constructor so that it happens asap, but that didn't work which is why I just moved the update to when we get an update from PickFirst.